feat: Allow experimental editor opt out

dcalhoun opened the pull request:
## Description
<!-- Describe the changes, why they are needed, and how they address the issue -->
Utilize the remote feature flag to enable a phased roll out. Allow opting out of
the experimental editor for a TBD period of time.

Ref CMM-291.

## Testing instructions
<!-- Consider listing specific testing steps for helping reviewers evaluate the efficacy of the changes -->
The experimental editor availability should operate much like any other remote
feature flag, where a remote value can be overridden by a local value controlled
by the user.

Users who enabled the now deprecated local-only feature flag should still see 
the experimental editor loaded. 

<!--

Consider testing the following *yourself* before requesting a review:

- Compatibility with WordPress.com sites and self-hosted Jetpack sites
- Both portrait and landscape orientations
- Light and dark modes
- Dynamic font sizes: larger, smaller and bold text
- High contrast mode
- Screen reader experience (e.g., VoiceOver)
- Languages with large words or with letters/accents not frequently used in English
- Right-to-left languages layout
- Multi-tasking: split view and slide over

-->

| Light | Dark |
| - | - |
| <img width="379" alt="Screenshot 2025-04-24 at 14 54 28" src="https://github.com/user-attachments/assets/75522f46-11aa-4358-89fa-727f54a52199" /> | <img width="380" alt="Screenshot 2025-04-24 at 14 54 36" src="https://github.com/user-attachments/assets/077e0398-91db-4790-aac5-93764cfe857a" /> |

dcalhoun committed:
feat: List experimental feature notes beneath all toggles

Communicate nuances about individual experimental features.

dcalhoun committed:
feat: Utilize remote GBK feature for experiments list

Deprecate the local-only feature flag, instead use the remote feature
flag to enable phased roll out.

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

dcalhoun committed:
feat: Request feedback via remote GBK flag

Ensure feedback is requested regardless if which flag is
disabled--current or deprecated.

dcalhoun committed:
feat: Reader app enables defaults to GutenbergKit

Mirror the existing default for the local feature flag.

dcalhoun committed:
refactor: Remove the local GutenbergKit feature flag

There is now a remote GutenbergKit feature flag. Given that local and
remote feature flags share the same storage key pattern, we can simply
rely upon the new, remote feature flag, and no migration is necessary.
